packages/sass-variables/_typography.scss

```diff
@@ -217,21 +275,41 @@ $puppet-type-code-small-font-weight: $puppet-type-weight-normal !default;
   font-size: $puppet-type-code-font-size;
   font-weight: $puppet-type-code-font-weight;
   line-height: $puppet-type-code-line-height;
+  background-color: $puppet-n100;
+  padding: 0 $puppet-common-spacing-base;
+  border-radius: 2px;
 }
 
 @mixin puppet-type-code-small() {
+  @include puppet-type-code();
   font-family: $puppet-type-code-small-font-family;
   font-size: $puppet-type-code-small-font-size;
   font-weight: $puppet-type-code-small-font-weight;
   line-height: $puppet-type-code-small-line-height;
 }
 
+// Same as puppet-type-code, but scale the font size to that of the surrounding
+// text (based on the ratio of default body & code font sizes)
+@mixin puppet-type-code-scaled() {
+  @include puppet-type-code();
+  font-size: #{$puppet-type-code-font-size / $puppet-type-body-font-size}em;
+  line-height: inherit;
```

Comment:
This is not perfect - this generates the correct size for body text (15px body vs. 16px code), and scales other things in a relatively sane way, but it gets maybe a little weird at smaller sizes, where the designed ratio between body & code sizes is different. This means that the "scaled" size inside some small text is not exactly the same as the dedicated `small` code size we use in the Code component, but I'm not sure how to address that in a better way. The example I modified on the Content page demonstrates all the scaled sizes - happy to adjust this approach if people have advice.